metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -166,7 +166,14 @@ func getResource(resourceName string) kubernetes.Resource {
 func getExtraWatchers(resourceName string, addResourceMetadata *metadata.AddResourceMetadataConfig) []string {
 	switch resourceName {
 	case PodResource:
-		extra := []string{NamespaceResource, NodeResource}
+		extra := []string{}
+		if addResourceMetadata.Node.Enabled() {
+			extra = append(extra, NodeResource)
+		}
+		if addResourceMetadata.Namespace.Enabled() {
```

Comment:
This made think of the scenario:
a) kubernetes.provider has add_resource.metadata.namespace.enabled: false and the module enabled: This means that the watcher will start. Always the module config is more specific correct?

Reply:
Yes, I believe the flow is provider > module > metricset.